navigation/src/androidMain/kotlin/com/freeletics/khonshu/navigation/NavHost.kt

```diff
@@ -131,9 +131,11 @@ private fun SystemBackHandling(snapshot: StackSnapshot, navigator: Navigator) {
 
     DisposableEffect(backPressedDispatcher, callback) {
         backPressedDispatcher.onBackPressedDispatcher.addCallback(callback)
+        backPressedDispatcher.onBackPressedDispatcher.addCallback(navigator.onBackPressedCallback)
```

Comment:
Theoretically we could combine these 2 but we'd need to split them again once we support predictive back (since that needs to be handled inside nav host) so I kept the existing code as is.